File: python/matching-brackets/matching_brackets.py
def is_paired(input_string):
    brackets = "".join(c for c in input_string if c in "()[]{}")

    while "()" in brackets or "[]" in brackets or "{}" in brackets:
        brackets = brackets.replace("()", "").replace("[]", "").replace("{}", "")
    
    return len(brackets) == 0

File: csharp/matching-brackets/MatchingBrackets.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

public static class MatchingBrackets
{
    public static bool IsPaired(string input)
    {
        var curatedInput = input
            .Where(c => c.Equals('(') || c.Equals(')') ||
                        c.Equals('[') || c.Equals(']') ||
                        c.Equals('{') || c.Equals('}'));

        var brackets = string.Concat(curatedInput);

        while (brackets.Contains("()") || brackets.Contains("[]") || brackets.Contains("{}"))
            brackets = brackets.Replace("()", "").Replace("[]", "").Replace("{}", "");

        return brackets.Length == 0;
    }
}
